desc/internal/util.go

```diff
@@ -1,13 +1,18 @@
 package internal
 
 import (
+	"math"
 	"unicode"
 	"unicode/utf8"
 )
 
 const (
-	// MaxTag is the maximum allowed tag number for a field.
-	MaxTag = 536870911 // 2^29 - 1
+	// MaxNormalTag is the maximum allowed tag number for a field in a normal message.
+	MaxNormalTag = 536870911 // 2^29 - 1
```

Comment:
Nit: I might switch this from MaxNormalTag/MaxTag to MaxTag/MaxTagMessageSetWireFormat since the message set max tag is the edge case, but more a matter of preference

Reply:
I wanted a name like `MaxTag` to be the actual maximum value -- the greater of the two -- because I thought that would be more clear.

Maybe `MaxNormalTag`, `MaxMessageSetTag`, and `MaxTag = MaxMessageSetTag`?

Reply:
Yea that works! Or keep as is, either way :-)

Reply:
In addition to adding `GetMaxTag(messageSet bool)`, I also left these other constants, but tweaked them per discussion above.